src/libsystemd-network/sd-dhcp-server.c

```diff
@@ -1397,3 +1479,59 @@ int sd_dhcp_server_set_relay_agent_information(
         free_and_replace(server->agent_remote_id, remote_id_dup);
         return 0;
 }
+
+int sd_dhcp_server_set_static_lease(
+                sd_dhcp_server *server,
+                const struct in_addr* address,
+                uint8_t *client_id,
+                size_t client_id_size) {
+        _cleanup_(dhcp_lease_freep) DHCPLease *lease = NULL, *old = NULL;
+        DHCPClientId c;
+        int r;
+
+        assert_return(server, -EINVAL);
+        assert_return(client_id, -EINVAL);
+        assert_return(client_id_size == ETH_ALEN, -EINVAL);
+        assert_return(!sd_dhcp_server_is_running(server), -EBUSY);
+        /* Static lease with an empty or omitted address is a valid entry,
+        * the server removes any static lease with the specified mac address. */
+        if(address == NULL || address->s_addr == 0) {
```

Comment:
Missing white space after `if`. Also, we use implicit cast from pointer to boolean, so
```
if (!address || address->s_addr == 0)
```